parquet-thrift/src/main/java/parquet/thrift/projection/FieldProjectionFilter.java

```diff
@@ -68,16 +72,26 @@ public FieldProjectionFilter(String filterDescStr) {
     }
   }
 
+    @Override
   public boolean isMatched(FieldsPath path) {
```

Comment:
Can tracking the paths that have matched or not be done where this is called? I think it would be better to not keep state since this class functions as a predicate.